src/lib/y2partitioner/widgets/lvm_vg_devices_selector.rb

```diff
@@ -0,0 +1,156 @@
+    class LvmVgDevicesSelector < Widgets::DevicesSelection
+      # Constructor
+      #
+      # @param controller [Actions::Controllers::LvmVg]
+      def initialize(controller)
+        @controller = controller
+        super()
+      end
+
+      # @see Widgets::DevicesSelection#selected
+      def selected
+        controller.devices_in_vg
+      end
+
+      # @see Widgets::DevicesSelection#selected_size
+      def selected_size
+        controller.vg_size
+      end
+
+      # @see Widgets::DevicesSelection#unselected
+      def unselected
+        controller.available_devices
+      end
+
+      # @see Widgets::DevicesSelection#select
+      def select(sids)
+        find_by_sid(unselected, sids).each do |device|
+          controller.add_device(device)
+        end
+      end
+
+      # @see Widgets::DevicesSelection#unselect
+      #
+      # @note Committed physical volumes cannot be unselected, see {#validate_unselect}.
+      def unselect(sids)
+        validate_unselect(sids)
+        sids -= controller.committed_devices.map(&:sid)
+
+        find_by_sid(selected, sids).each do |device|
+          controller.remove_device(device)
+        end
+      end
+
+      # Validates the selected physical volumes
+      # @macro seeAbstractWidget
+      #
+      # @see #selected_devices_validation
+      # @see #size_validation
+      #
+      # @note An error popup is shown when there is some error in selected devices.
+      #
+      # @return [Boolean]
+      def validate
+        selected_devices_validation && size_validation
+      end
+
+    private
+
+      # @return [Actions::Controllers::LvmVg]
+      attr_reader :controller
+
+      # Validates that at least one physical volume was added to the volume group
+      #
+      # @note An error popup is shown when no physical volume was added.
+      #
+      # @return [Boolean]
+      def selected_devices_validation
+        return true if controller.devices_in_vg.size > 0
+
+        Yast::Popup.Error(_("Select at least one device."))
+        false
+      end
+
+      # Validates that the volume group size is enough for all its logical volumes
+      #
+      # @note An error popup is shown when the size is not enough.
+      #
+      # @return [Boolean]
+      def size_validation
+        return true if controller.vg_size >= controller.lvs_size
+
+        Yast::Popup.Error(
+          format(
+            _("The volume group size cannot be less than %s."), controller.lvs_size
+          )
+        )
+
+        false
+      end
+
+      # Validates that unselected phyical volumes can be removed
+      #
+      # @note A physical volume cannot be removed if it is already committed. An error
+      #   popup is shown when a committed physical volume is tried to be removed.
+      #
+      # @see Actions::Controllers::LvmVg#committed_devices
+      #
+      # @param sids [Array<Integer>]
+      # @return [Boolean]
+      def validate_unselect(sids)
```

Comment:
NP: not sure whether it makes sense to call this method `validate_xxx` and make it return a boolean. It's only real usage is displaying messages to the user but is not used to interrupt the option or something like that (in fact, the return value is ignored).

I can live with it, but it's kind of confusing.

Reply:
Yes, in this case it is only a check, not a validation.